source/location_access/locacc_ice_cavern.cpp

```diff
@@ -63,6 +66,12 @@ void AreaTable_Init_IceCavern() {
                 LocationAccess(ICE_CAVERN_GS_SPINNING_SCYTHE_ROOM, { [] { return HookshotOrBoomerang; } }),
                 LocationAccess(ICE_CAVERN_GS_HEART_PIECE_ROOM, { [] { return BlueFire && HookshotOrBoomerang; } }),
                 LocationAccess(ICE_CAVERN_GS_PUSH_BLOCK_ROOM, { [] { return BlueFire && HookshotOrBoomerang; } }),
+                LocationAccess(ICE_CAVERN_PUSH_BLOCK_ROOM_RED_RUPEE_1,
+                               { [] { return SongOfTime || CanUse(BOOMERANG); } }),
```

Comment:
SongOfTime does not include the Ocarina check, only the buttons and song
```suggestion
                               { [] { return CanPlay(SongOfTime) || CanUse(BOOMERANG); } }),
```